src/common/libflux/ev_buffer_write.c

```diff
@@ -40,7 +42,16 @@ static void buffer_write_cb (struct ev_loop *loop, ev_io *iow, int revents)
         if (flux_buffer_read_to_fd (ebw->fb, ebw->fd, -1) < 0)
             return;
 
-        if (!flux_buffer_bytes (ebw->fb))
+        if (!flux_buffer_bytes (ebw->fb) && ebw->eof) {
+            if (close (ebw->eof) < 0)
```

Comment:
should be ```close (ebw->fd)```?

Reply:
Yipes, what a ridiculously ridiculous typo. Funny thing is in my `reactorcat` test this actually worked because it resolved to `close (1)` or `close (STDOUT_FILENO)`. (Luckily would be caught by the unit tests, but nice catch @chu11!)